src/components/HotspotEditorModal/HotspotEditorDataSourceTab/HotspotEditorDataSourceTab.jsx

```diff
@@ -0,0 +1,200 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Edit16 } from '@carbon/icons-react';
+import { MultiSelect } from 'carbon-components-react';
+
+import DataSeriesFormItemModal from '../../CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItemModal';
+import List from '../../List/List';
+import Button from '../../Button/Button';
+import { settings } from '../../../constants/Settings';
+
+const { iotPrefix } = settings;
+
+const propTypes = {
+  title: PropTypes.string.isRequired,
+  cardConfig: PropTypes.shape({
```

Comment:
I still don't think the complete `cardConfig` is something that a tab living inside a modal should know about. IMO the modal should know about the card config and the internal sub components should know as little as possible (Law of Demeter). I think that from the cardConfig object this tab should only receive  

1. The attributes array of the currently selected hotspot 
2. Thresholds array

However, since the `DataSeriesFormItemModal` is designed to require the complete `cardConfig` for its handleDataItemEdit we still need it. I guess this is a consequence of the unusual design decision to have a data input modal inside another data input modal.

Reply:
We can refactor this at some point to let the hotspot modal pop the DataSeriesFormItemModal.